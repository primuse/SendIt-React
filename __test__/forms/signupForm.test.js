import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import ConnectedSignupForm, { SignupForm }
  from '../../src/components/forms/signUpForm.jsx';

const onChange = jest.fn();
const signupUser = jest.fn();
const props = {
  auth: {
    isLoading: false
  }
};
const mockStore = configureStore([thunk]);
const store = mockStore(props);
let wrapper;

describe('Signup form', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Provider store={store}>
        <ConnectedSignupForm />
      </Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Signup form', () => {
  beforeEach(() => {
    wrapper = mount(<SignupForm {...props}
      onChange={onChange} signupUser={signupUser} />);
  });
  afterEach(() => {
    wrapper.unmount();
  });
  it('should change state', () => {
    const input = wrapper.find('#firstName');
    input.instance().value = 'Tiku';
    input.simulate('change');
    expect(wrapper.state().firstName).toEqual('Tiku');
  });
  it('should simulate the onSubmit', () => {
    wrapper.find('form').simulate('submit');
    const input1 = wrapper.find('input').at(0);
    const input2 = wrapper.find('input').at(1);
    const input3 = wrapper.find('input').at(2);
    const input4 = wrapper.find('input').at(3);
    input1.simulate('change', { target: { value: 'Tiku' } });
    input2.simulate('change', { target: { value: 'Okoye' } });
    input3.simulate('change', { target: { value: 'cim@gmail.com' } });
    input4.simulate('change', { target: { value: 'cim23000' } });
    expect(signupUser).toHaveBeenCalled();
  });
});
