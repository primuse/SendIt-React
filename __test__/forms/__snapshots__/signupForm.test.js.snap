// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup form renders correctly 1`] = `
<form
  id="signupForm"
  onSubmit={[Function]}
>
  <div
    className="column is-two form-row"
  >
    <input
      id="firstName"
      name="firstName"
      onChange={[Function]}
      placeholder="First name"
      required={true}
      type="text"
      value=""
    />
    <input
      id="lastName"
      name="lastName"
      onChange={[Function]}
      placeholder="Last name"
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="column is-two form-row"
  >
    <input
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      required={true}
      type="email"
      value=""
    />
    <input
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
  </div>
  <button
    className="btn lg is-outlined"
    type="submit"
  >
    Signup
  </button>
</form>
`;
